.golangci.yml

```
@@ -15,6 +15,11 @@ issues:
   # When true it disables revive's comment warnings, and we want them enabled.
   exclude-use-default: false
 
+  exclude-rules:
+    - linters:
+        - funlen
```

```suggestion
        - funlen
        - forcetypeassert
```

WDYT? Instead of using `nolint` every time

---

Hhmm I think the linter is useful in general, it's just that in these cases we know it must be this type, so I'd rather `nolint` it.

We could also just do:
```golang
cp, _ := p.(*common.Page)
kb := cp.Keyboard
```

Or to be safe:
```golang
cp, ok := p.(*common.Page)
require.True(t, ok)
kb := cp.Keyboard
```

But that's just noise for tests.

I don't have a strong preference, but the `nolint` doesn't bother me. :man_shrugging: Let me know if you strongly prefer another approach and I'll change it.

---

I considered that this linter is essential for the application code, but in the tests, I can't imagine the cases for that.

The issue with the `nolint` is that it's also a kind of code smell, so if there is a simple workaround like you mentioned before, it's better to use it to avoid `nolint`.

Regarding the suggested options, I'll vote for the safer one (with the usage of `require`). Why? Simply because even if this happens (during refactoring or so), we'll save some time for the future generations of the developers (or us) because they don't need investigating that `p` is not more the `*common.Page` or read panic output (as it is right now).

---

Sure, a couple more lines are worth avoiding the panic. Resolved in latest force push (hope you don't mind, wanted to keep this clean and ready for merging).

---

Jus a small nitpick :see_no_evil: 

If we're changing tests to using `require.True(t, ok)` I believe there should not be `forcetypeassert` in the `exclude-rules`

---

Yeah, it's not, I didn't want to disable it globally.

---

:man_facepalming: still not here with the Github UI. I was looking for my suggestion and thought that it was committed. Shame on me :sweat_smile: